File: M/1572-matrix-diagonal-sum/main.go
package main

import "fmt"

func diagonalSum(mat [][]int) int {
	size := len(mat)
	sum := 0
	for i := 0; i < size; i++ {
		sum += mat[i][i]
		sum += mat[size-1-i][i]
	}
	if size%2 == 1 {
		sum -= mat[size/2][size/2]
	}
	return sum
}

func main() {
	fmt.Println("a")
}

File: M/1572-matrix-diagonal-sum/main.rs
struct Solution {}

impl Solution {
  pub fn diagonal_sum(mat: Vec<Vec<i32>>) -> i32 {
    let mut sum = 0;
    let n = mat.len();
    for i in 0..n {
      sum += mat[i][i];
      sum += mat[i][n - i - 1];
    }
    if n % 2 == 1 {
      sum -= mat[n / 2][n / 2];
    }
    sum
  }
}
